File: tiny-get-jwt.rb
require 'openssl'
require 'jwt'  # https://rubygems.org/gems/jwt
require 'pathname'

private_pem_name = ARGV[0]
private_pem = if Pathname.new(private_pem_name).absolute?
                File.read(private_pem_name)
              else
                warn <<EOM
In .gh-api-examples.conf, a relative path is set to private_pem_file.

   #{private_pem_name}

It is regarded as a relative path from your home directory, but it is recommended to set an absolute path to avoid confusion.

   #{ENV.fetch("HOME")}/#{private_pem_name}
EOM
                File.read("#{Dir.home}/#{private_pem_name}")
              end
private_key = OpenSSL::PKey::RSA.new(private_pem)


offset_time_by = 0
n = (Time.now.to_i + offset_time_by) + (10 * 60)

payload = {
  # iat = issued at time
  iat: Time.now.to_i,
  # exp = JWT expiration time (10 minute maximum)
  exp: n,
  # GitHub App ID identifier
  iss: ARGV[1].to_i
}

jwt = JWT.encode(payload, private_key, "RS256")
puts jwt

File: tiny-get-jwt.py
import jwt
import time 
import sys

# Get PEM file path
if len(sys.argv) > 1:
    pem = sys.argv[1]
else:
    pem = input("Enter path of private PEM file: ")    

# Open PEM
with open(pem, 'r') as pem_file:
    signing_key = pem_file.read()
    
payload = {
    # Issued at time
    'iat': int(time.time()),
    # JWT expiration time (10 minutes maximum)
    'exp': int(time.time()) + 600, 
    # GitHub App's identifier
    'iss': 'YOUR_APP_ID' 
}
    
# Create JWT
encoded_jwt = jwt.encode(payload, signing_key, algorithm='RS256')
     
print(encoded_jwt())
